### PR TITLE
FHG-1982 Improve code

### PR DESCRIPTION
**Pull request by MartinBelton-gov**
FHG-1982 Improve code

### CODE REVIEW

File: src/FamilyHubs.ReferralUi.Ui/Extensions/StringExtensions.cs [hunk 1]
```diff
@@ -8,6 +8,6 @@ public static string ToSentenceCase(this string input)
             return input;
 
         string sentence = input.ToLower();
-        return sentence[0].ToString().ToUpper() + sentence.Substring(1);
+        return string.Concat(sentence[0].ToString().ToUpper(), sentence.AsSpan(1));
```
Comment (lazcool) [inline src/FamilyHubs.ReferralUi.Ui/Extensions/StringExtensions.cs:11]: `Concat` works, but is there a reason why we can't use `$""`?

Comment (MartinBelton-gov) [reply src/FamilyHubs.ReferralUi.Ui/Extensions/StringExtensions.cs]: I have altered the code as you have suggested. That is what sonarqube suggested

File: src/FamilyHubs.ReferralUi.Ui/Filters/ApiExceptionFilterAttribute.cs [hunk 2]
```diff
@@ -31,7 +31,7 @@ public override void OnException(ExceptionContext context)
     private void HandleException(ExceptionContext context)
     {
         Type type = context.Exception.GetType();
-        if (_exceptionHandlers.ContainsKey(type))
+        if(_exceptionHandlers.TryGetValue(type, out _))
```
Comment (lazcool) [inline src/FamilyHubs.ReferralUi.Ui/Filters/ApiExceptionFilterAttribute.cs:34]: Just curious, is this something Sonar complained about?

File: src/FamilyHubs.ReferralUi.Ui/Services/Api/LocalOfferClientService.cs [hunk 3]
```diff
@@ -21,79 +20,103 @@ public LocalOfferClientService(HttpClient client)
...
+    public async Task<PaginatedList<ServiceDto>> GetLocalOffers(LocalOfferFilter filter) // string? serviceType, string status, int? minimum_age, int? maximum_age, int? given_age, string? districtCode, double? latitude, double? longtitude, double? proximity, int pageNumber, int pageSize, string text, string? serviceDeliveries, bool? isPaidFor, string? taxonmyIds, string? languages, bool? canFamilyChooseLocation)
     {
-        if (string.IsNullOrEmpty(status))
-            status = "active";
+        if (string.IsNullOrEmpty(filter.Status))
+            filter.Status = "active";
 
-        string url = string.Empty;
-        if (latitude != null && longtitude != null)
+        string url = GetPositionUrl(filter.ServiceType, filter.Latitude, filter.Longtitude, filter.Proximity, filter.Status, filter.PageNumber, filter.PageSize);
```
Comment (lazcool) [inline src/FamilyHubs.ReferralUi.Ui/Services/Api/LocalOfferClientService.cs:28]: Fine, but StringBuilder would reduce memory churn.

File: tests/FamilyHubs.ReferralUi.UnitTests/Services/WhenUsingLocalOfferClientService.cs [hunk 4]
```diff
@@ -1,9 +1,11 @@
-﻿using FamilyHubs.ReferralUi.Ui.Services.Api;
+﻿using FamilyHubs.ReferralUi.Ui.Models;
+using FamilyHubs.ReferralUi.Ui.Services.Api;
 using FamilyHubs.ServiceDirectory.Shared.Dto;
 using FamilyHubs.SharedKernel;
 using FluentAssertions;
 using Moq;
 using Newtonsoft.Json;
+using System.Drawing.Printing;
```
Comment (lazcool) [inline tests/FamilyHubs.ReferralUi.UnitTests/Services/WhenUsingLocalOfferClientService.cs:8]: Is this needed?

File: src/FamilyHubs.ReferralUi.Ui/Filters/ApiExceptionFilterAttribute.cs (hunk 2 above)
Comment (MartinBelton-gov) [reply src/FamilyHubs.ReferralUi.Ui/Filters/ApiExceptionFilterAttribute.cs:34]: Yes

File: src/FamilyHubs.ReferralUi.Ui/Services/Api/LocalOfferClientService.cs (hunk 3 above)
Comment (MartinBelton-gov) [reply src/FamilyHubs.ReferralUi.Ui/Services/Api/LocalOfferClientService.cs:28]: Uses String Builder now

File: src/FamilyHubs.ReferralUi.Ui/Pages/Index.cshtml.cs [hunk 5]
```diff
@@ -5,18 +5,15 @@ namespace FamilyHubs.ReferralUi.Ui.Pages;
...
     public IndexModel(IConfiguration configuration, ILogger<IndexModel> logger)
     {
-        _logger = logger;
         IsReferralEnabled = configuration.GetValue<bool>("IsReferralEnabled");
     }
 
     public void OnGet()
     {
-
+        //Standard GET method for page
```
[the remark this answers is not in the document]
Comment (MartinBelton-gov) [reply src/FamilyHubs.ReferralUi.Ui/Pages/Index.cshtml.cs:17]: Remove Get methods

File: tests/FamilyHubs.ReferralUi.UnitTests/Services/WhenUsingLocalOfferClientService.cs (hunk 4 above)
Comment (MartinBelton-gov) [reply tests/FamilyHubs.ReferralUi.UnitTests/Services/WhenUsingLocalOfferClientService.cs:8]: Nope removed it

**Pull request merged**